components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx

```diff
@@ -27,21 +30,30 @@ export default class PermissionSchemesSettings extends React.PureComponent {
             loading: true,
             loadingMore: false,
             page: 0,
+            phase2MigrationIsComplete: false,
         };
     }
 
     static defaultProps = {
         schemes: [],
     };
 
-    componentDidMount() {
-        this.props.actions.loadSchemes('team', 0, PAGE_SIZE).then((schemes) => {
+    async componentWillMount() {
+        let schemes;
+        let phase2MigrationIsComplete = true; // Assume migration is complete unless HTTP status code says otherwise.
+        try {
+            schemes = await this.props.actions.loadSchemes('team', 0, PAGE_SIZE);
+            if (schemes.error.status_code === PHASE_2_MIGRATION_IMCOMPLETE_STATUS_CODE) {
+                phase2MigrationIsComplete = false;
```

Comment:
I'm not sure if it makes sense to continue the execution of the method in this point. I would execute the setState and return here. Probably haven't a different effect in the execution, but is clearer. Not an stopper at all.